views/js/qtiCreator/plugins/panel/outcomeEditor.js

```diff
@@ -136,11 +215,45 @@ define([
 
                             //save to model
                             outcome.attr('interpretation', value);
+                        },
+                        externalScored : function(outcome, value){
+                            //Turn off scoring trait validation if externalScored is not human
+                            isScoringTraitValidaitonEnabled = (value === externalScoredOptions.human);
+                            /**
+                             * Save to model
+                             *
+                             * Removes the `externalScored` attribute from outcome when `none` is selected.
+                             * Attaches scoring trait warning tooltips when `externalScored` is `human`
+                             *
+                             */
+                            if(value === externalScoredOptions.none) {
+                                outcome.removeAttr('externalScored');
+                                disposeScoringTraitWarningTooltip($outcomeValueMinimum);
+                                disposeScoringTraitWarningTooltip($outcomeValueMaximum);
+                            } else if( value === externalScoredOptions.human) {
+                                outcome.attr('externalScored', value);
+                                attachScoringTraitWarningTooltip($outcomeValueMinimum);
+                                attachScoringTraitWarningTooltip($outcomeValueMaximum);
+                                // shows tooltips in case of invalid value
+                                showScoringTraitWarningOnInvalidValue();
+                            } else {
+                                outcome.attr('externalScored', value);
+                                disposeScoringTraitWarningTooltip($outcomeValueMinimum);
+                                disposeScoringTraitWarningTooltip($outcomeValueMaximum);
```

Comment:
Could you refactor the 3 paths to avoid duplication of code?

All 3 are doing almost the same:
- 1 is removing the attr, while the 2 others are setting it
- 1 is attaching the tooltip, while the 2 others are removing it